### PR TITLE
guarded suspension pattern #69

### PR DESCRIPTION
**Pull request by robertt240**
pull request for guarded suspension pattern

### CODE REVIEW

File: guarded-suspension/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+folder: guarded-suspension
+permalink: /patterns/guarded-suspension/
+categories: Concurrency
+tags:
+ - Java
+ - Difficulty-Beginner
+---
+
+## Intent
+Use Guareded suspension pattern to handle a situation when you want to execute a method on object which is not in a proper state.
```
Comment (iluwatar) [inline guarded-suspension/README.md]: Typo

File: guarded-suspension/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+![Guarded Suspension diagram](./etc/guarded-suspension.png)
+
+## Applicability
+Use Guareded Suspension pattern when:
```
Comment (iluwatar) [inline guarded-suspension/README.md]: Typo

File: guarded-suspension/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+* the developer knows that the method execution will be blocked for a finite period of time
+
```
Comment (iluwatar) [inline guarded-suspension/README.md]: I would add related patterns section, with reference to Balking pattern

File: guarded-suspension/pom.xml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    <groupId>com.iluwatar</groupId>
+    <artifactId>java-design-patterns</artifactId>
+    <version>1.14.0-SNAPSHOT</version>
+  </parent>
+  <packaging>jar</packaging>
+  <artifactId>guarded-suspension</artifactId>
+  <dependencies>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
```
Comment (iluwatar) [inline guarded-suspension/pom.xml]: Scope should be test

File: guarded-suspension/src/main/java/com/iluwatar/guarded/suspension/GuardedQueue.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+
+public class GuardedQueue {
+  private static final Logger LOGGER = LoggerFactory.getLogger(GuardedQueue.class);
+  private Queue<Integer> sourceList;
```
Comment (iluwatar) [inline guarded-suspension/src/main/java/com/iluwatar/guarded/suspension/GuardedQueue.java]: Make it final, if it only changes in object construction

**Comment by robertt240** [comment]
Finally, I introduced changes which you suggested. I have no idea why CI failed, because when i run the same maven command at my local machine it succeed.

**Comment by iluwatar** [comment]
@robertt240 I think the problem with the build is that your version number is lagging behind. The version at master branch is `1.15.0-SNAPSHOT` and you should have the same.

**Comment by robertt240** [comment]
Okay, it seems that I finally managed to rebase on 1.15 version and I am ready to review.

**Comment by iluwatar** [comment]
@robertt240 looks good now. Thank you very much for the contribution!

**Comment by iluwatar** [comment]
@all-contributors please add @robertt240 for code

**Pull request merged**